Common/Chairloader/ConfigNode.h

```diff
@@ -0,0 +1,247 @@
+
+inline NodeType toNodeType(std::string_view s) {
+    if      (s=="string")   return NodeType::String;
+    else if (s=="bool")     return NodeType::Bool;
+    else if (s=="int")      return NodeType::Int;
+    else if (s=="uint")     return NodeType::Uint;
+    else if (s=="int64")    return NodeType::Int64;
+    else if (s=="uint64")   return NodeType::Uint64;
+    else if (s=="float")    return NodeType::Float;
+    else if (s=="xmlnode")  return NodeType::XmlNode;
+    else if (s=="enum")     return NodeType::Enum;
+    else                    return NodeType::Unknown;
+}
+
+inline std::string toString(NodeType type) {
+    switch(type) {
+        case NodeType::String:   return "string";
+        case NodeType::Bool:     return "bool";
+        case NodeType::Int:      return "int";
+        case NodeType::Uint:     return "uint";
+        case NodeType::Int64:    return "int64";
+        case NodeType::Uint64:   return "uint64";
+        case NodeType::Float:    return "float";
+        case NodeType::XmlNode:  return "xmlnode";
+        case NodeType::Enum:     return "enum";
+        default:                 return "unknown";
+    }
+}
+
+
+class ConfigNode {
+public:
+    typedef std::variant<int, unsigned int, float, bool, int64_t, uint64_t, std::string, pugi::xml_node> ValueType;
+
+    ConfigNode(pugi::xml_node node)
+    : m_node(node)
+    {
+        if(m_node){
+            m_name = m_node.name();
+            m_displayName = m_node.attribute("displayName").as_string();
+            m_description = m_node.attribute("description").as_string();
+            m_type = m_node.attribute("type").as_string();
+            m_nodeType = toNodeType(m_type);
+        }
+        else {
+            m_name = "";
+            m_displayName = "";
+            m_description = "";
+            m_type = "";
+            m_nodeType = NodeType::XmlNode;
+        }
+    }
+
+    ConfigNode() : m_node() {
+        m_name = "";
+        m_displayName = "";
+        m_description = "";
+        m_type = "";
+        m_nodeType = NodeType::XmlNode;
+    }
+
+    ConfigNode operator[](const char* name) {
+        return ConfigNode(m_node.child(name));
+    }
+
+    ConfigNode get(const char* name) {
+        return ConfigNode(m_node.child(name));
+    }
+
+    ConfigNode create(const char* name, NodeType type) {
+        if (m_node.child(name)) {
+            throw std::runtime_error("Node already exists");
+        }
+        auto node = m_node.append_child(name);
+        node.append_attribute("type") = toString(type).c_str();
+        node.append_child(pugi::node_pcdata).set_value("");
+        return ConfigNode(node);
+    }
+
+    ConfigNode getOrCreate(const char* name, NodeType type) {
+        if (m_node.child(name)) {
+            return ConfigNode(m_node.child(name));
+        }
+        return create(name, type);
+    }
+
+    bool exists() const {
+        return m_node;
+    }
```

Comment:
I think calling this `IsValid` would make more sense. Otherwise, how can a config node not exist if you call a method on it?